### PR TITLE
Issue #1033 - Fixed requirements file parsing when it contains pip options

### PR DESCRIPTION
**Pull request by mbacicc**
Fixed bandersnatch crash when requirements file contains [pip options.](https://pip.pypa.io/en/latest/reference/requirements-file-format/#supported-options)

Test result before the PR:

```
self = <[AttributeError("'Requirement' object has no attribute 'name'") raised in repr()] Requirement object at 0x7f230a96c5e0>
requirement_string = '--extra-index-url https://self-hosted-foo.netname/simple --trusted-host self-hosted-foo.netname'

    def __init__(self, requirement_string: str) -> None:
        try:
            req = REQUIREMENT.parseString(requirement_string)
        except ParseException as e:
>           raise InvalidRequirement(
                f'Parse error at "{ requirement_string[e.loc : e.loc + 8]!r}": {e.msg}'
            )
E           packaging.requirements.InvalidRequirement: Parse error at "'--extra-'": Expected W:(0-9A-Za-z)

venv/lib/python3.8/site-packages/packaging/requirements.py:104: InvalidRequirement
```

### CODE REVIEW

File: src/bandersnatch_filter_plugins/allowlist_name.py [hunk 1]
```diff
@@ -121,7 +121,7 @@ def _parse_package_lines(package_lines: List[str]) -> Set[Requirement]:
     filtered_requirements: Set[Requirement] = set()
     for package_line in package_lines:
         package_line = package_line.strip()
-        if not package_line or package_line.startswith("#"):
+        if not package_line or package_line.startswith("#") or package_line.startswith("-"):
```
Comment (cooperlees) [inline src/bandersnatch_filter_plugins/allowlist_name.py:124]:
```suggestion
        if not package_line or package_line.startswith(("#", "-"))):
```
Pretty sure this works - I only tested in 3.10.6 tho ... Don't change if 3.8 can't handle this.

File: CHANGES.md [hunk 2]
```diff
@@ -1,3 +1,9 @@
+# PR #1231
+
+## Bug Fixes
+
+- Fixed requirements file parsing when it contains pip options
```
Comment (cooperlees) [inline CHANGES.md:5]:
```suggestion
# Unreleased

## Bug Fixes

- Fixed requirements file parsing when it contains pip options `PR #1231`
```

**Review by cooperlees — APPROVED** [review]
SGTM - I'll try merge some small fixes here too.

**Review by cooperlees — CHANGES_REQUESTED** [review]
Bummer - please format the files with black.

File: src/bandersnatch_filter_plugins/allowlist_name.py [hunk 3]
```diff
@@ -121,7 +121,7 @@ def _parse_package_lines(package_lines: List[str]) -> Set[Requirement]:
     filtered_requirements: Set[Requirement] = set()
     for package_line in package_lines:
         package_line = package_line.strip()
-        if not package_line or package_line.startswith("#"):
+        if not package_line or package_line.startswith(("#", "-"))):
```
Comment (cooperlees) [inline src/bandersnatch_filter_plugins/allowlist_name.py:124]:
```suggestion
        if not package_line or package_line.startswith(("#", "-")):
```
Seems editing python via the web is evil / I am useless :P

File: src/bandersnatch/tests/plugins/test_allowlist_name.py [hunk 4]
```diff
@@ -458,6 +458,41 @@ def test__filter__find_files(self) -> None:
...
+        }
+        mirror._filter_packages()
+        self.assertEqual({"foo": ""}, mirror.packages_to_sync)
+
+    def test__filter__requirements__pip__options(self) -> None:
+        absolute_file_path = Path(self.tempdir.name) / "requirements.txt"
+        with open(absolute_file_path, "w") as fh:
+            fh.write(
+                """\
+--extra-index-url https://self-hosted-foo.netname/simple --trusted-host self-hosted-foo.netname
```
Comment (cooperlees) [inline src/bandersnatch/tests/plugins/test_allowlist_name.py:485]:
```suggestion
--extra-index-url https://self-hosted-foo.netname/simple
--trusted-host self-hosted-foo.netname
```

**Comment by mbacicc** [comment]
Sorry, my fault, didn't configure pre-commits.

Seems everything resolved now.

**Review by cooperlees — APPROVED** [review]

**Pull request merged**